assets/shared/helpers/player.js

```diff
@@ -0,0 +1,552 @@
+const VIDEO_TYPE = 'video';
+const VIDEOPRESS_TYPE = 'videopress';
+const YOUTUBE_TYPE = 'youtube';
+const VIMEO_TYPE = 'vimeo';
+
+const players = {
+	[ VIDEO_TYPE ]: {
+		/**
+		 * Initialize the player.
+		 *
+		 * @param {HTMLVideoElement} element The player element.
+		 *
+		 * @return {Promise<HTMLVideoElement>} The video player through a promise.
+		 */
+		initializePlayer: ( element ) => {
+			// Return that it's ready when it can get the video duration.
+			if ( ! isNaN( element.duration ) ) {
+				return Promise.resolve( element );
+			}
+
+			return new Promise( ( resolve ) => {
+				element.addEventListener( 'durationchange', () => {
+					resolve( element );
+				} );
+			} );
+		},
+
+		/**
+		 * Get the video duration.
+		 *
+		 * @param {HTMLVideoElement} player The player element.
+		 *
+		 * @return {Promise<number>} The duration of the video in seconds through a promise.
+		 */
+		getDuration: ( player ) => Promise.resolve( player.duration ),
+
+		/**
+		 * Set the video to a current time.
+		 *
+		 * @param {HTMLVideoElement} player  The player element.
+		 * @param {number}           seconds The video time in seconds to set.
+		 *
+		 * @return {Promise} A promise that resolves if the video was set to a current time successfully.
+		 */
+		setCurrentTime: ( player, seconds ) => {
+			try {
+				player.currentTime = seconds;
+				return Promise.resolve();
+			} catch ( e ) {
```

Comment:
can `player.currentTime` assignment raise an error?

Reply:
Yep! Testing quickly, I could reproduce an error trying to set a string as `currentTime`. And maybe the browsers can implement other types of error in the future? Like setting a time bigger than the video or something like that?

Reply:
Good to know!